pkg/kubelet/kubelet.go

```diff
@@ -1747,6 +1750,24 @@ func (kl *Kubelet) syncLoop(updates <-chan kubetypes.PodUpdate, handler SyncHand
 	housekeepingTicker := time.NewTicker(housekeepingPeriod)
 	defer housekeepingTicker.Stop()
 	plegCh := kl.pleg.Watch()
+
+	var watchdogCh <-chan time.Time
+	var watchdogTicker *time.Ticker
+
+	// If a value for WatchdogSec is defined in our systemd unit file,
+	// enable detection of hung kubelet processes.  Require *at least* 2
+	// genuinely missed heartbeats before recovery action is taken
+	watchdogEnv, err := strconv.ParseInt(os.Getenv("WATCHDOG_USEC"), 10, 64)
```

Comment:
Should we add cases to check where the env var has an unexpected value? This implementation will just ignore it and don't log anything.

Reply:
After reading here: https://www.freedesktop.org/software/systemd/man/sd_watchdog_enabled.html, it looks like `WATCHDOG_PID` is related as well. Should we handle that?

Reply:
Hi, this is Marc, @beekhof asked me to continue with this. PR. Thanks for your feedback so far! I will update the PR soonish, just need to get a bit more familiar with the kubelet code.
You are right with the comments here, I will address them by using `func SdWatchdogEnabled(unsetEnvironment bool) (time.Duration, error)` of the watchdog lib Andrew already introduced, see https://github.com/beekhof/kubernetes/blob/d462e1e8d7e45894bf2abb7dc5d939f1bd3cae68/vendor/github.com/coreos/go-systemd/daemon/watchdog.go#L35